File: files/CH-117.py
import pandas as pd

path = "CH-117 Add Index Column.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1)
test = pd.read_excel(path, usecols="E:G", skiprows=1).rename(columns=lambda x: x.replace(".1", ""))

result = input.assign(index = input.groupby("Stock").cumcount() + 1)
print(result.equals(test))  # True

File: files/100-199/117/CH-117.R
library(tidyverse)
library(readxl)

path = "files/CH-117 Add Index Column.xlsx"
input = read_excel(path, range = "B2:C13")
test  = read_excel(path, range = "E2:G13")

result = input %>%
  mutate(index = row_number(), .by = Stock)

all.equal(result, test)
#> [1] TRUE